jsonlinter.py

```diff
@@ -0,0 +1,98 @@
+# python3 lintjson.py
+# William Rehwinkel / obiesource, 2022
+
+import sys
+import os
+import json
+import traceback
+
+def getJsonFiles():
+    # list all files recursively borrowed from https://pythonexamples.org/python-get-list-of-all-files-in-directory-and-sub-directories/
+    filelist = list()
+    path = "./"
+    
+    for root, dirs, files in os.walk(path):
+	    for file in files:
+            #append the file name to the list
+		    filelist.append(os.path.join(root,file))
+    
+    #print all the file names
+    jsonFiles = list()
+    
+    for name in filelist:
+        if name.lower().endswith(".json"):
+            jsonFiles.append(name)
+    return jsonFiles
+
+MEMBERS_REQUIRED = {
+        "name": "",
+        "pronouns": "",
+        "class": "",
+        "socials": list(),
+        "bio": "",
+        "projects": list()
+        }
+PROJECTS_REQUIRED = {
+        "name": "",
+        "description": "",
+        "website": ""
+        }
+
+def lintJson(filename):
+    print("Checking file " + filename)
+    with open(filename) as file:
+        try: js = json.load(file)
+        except Exception as e:
+            print("Error while parsing " + filename)
+            traceback.print_tb(e)
+            return True
+
+        failed = False
+        if filename.startswith("./members/"):
+            # check for required keys
+            for k in MEMBERS_REQUIRED:
+                t = MEMBERS_REQUIRED[k]
```

Comment:
Fun python thing: iterables aren't always served as single items. With dictionaries you can do
```suggestion
            for k, t in MEMBERS_REQUIRED.items():
```
with the same effect.

I think it's more efficient, because `.items()` iterates over the pairs without ever searching for any given key, but I don't know how it's implemented.

Reply:
People also usually use `k, v` but I'm not sweating the use of `t` short for "type". You could also do
```suggestion
            for k, v in MEMBERS_REQUIRED.items:
                t = type(v)
```
and have the value stored in `t` _actually_ be the type. Wicked!